utils/prof_utils.py

```diff
@@ -0,0 +1,1150 @@
+
+    #W50 root errors
+    err_50_1 = error_in_x_pos(pcov, popt, W50_roots[0])
+    err_50_2 = error_in_x_pos(pcov, popt, W50_roots[-1])
+    W50_e = np.sqrt(err_50_1**2 + err_50_2**2)
+
+    #Wscat root errors
+    err_scat_1 = error_in_x_pos(pcov, popt, Wscat_roots[0])
+    err_scat_2 = error_in_x_pos(pcov, popt, Wscat_roots[-1])
+    Wscat_e = np.sqrt(err_scat_1**2 + err_scat_2**2)
+
+    #Weq errors - using covariance formula
+    on_pulse_less = (on_pulse - std).clip(min=0)
+    spline0 = UnivariateSpline(x, on_pulse_less, s=0)
+    integral = spline0.integral(0, len(profile)-1)
+    dwdint = 1/max(on_pulse)**2
+    dwdmax = -integral/max(on_pulse)**2
+    int_e = abs(integral/max(on_pulse - std) - integral/max(on_pulse))
+    max_e = std
+    Weq_e = np.sqrt( dwdint**2 * int_e**2 + dwdmax**2 * max_e**2 + 2*dwdint*dwdmax*int_e*max_e )
+
+    return [W10, W50, Weq, Wscat, W10_e, W50_e, Weq_e, Wscat_e]
+
+#---------------------------------------------------------------
+def est_sn_from_prof(prof_data, period, alpha=3.):
+    """
+    Estimates the signal to noise ratio from a pulse profile
+    Based on code oringally writted by Nick Swainston.
+
+    Parameters:
+    -----------
+    prof_data: string
+        A list of floats that contains the pulse profile
+    period: float
+        The pulsar's period in ms
+    alpha: float
+        OPTIONAL - The alpha value to be used in sigmaClip(). Default: 3
+
+    Returns:
+    --------
+    [sn, sn_e, scattered]
+    sn: float
+        The estimated signal to noise ratio
+    u_sn: float
+        The uncertainty in sn
+    scattered: boolean
+        When true, the profile is highly scattered
+    """
+    #centre the profile around the max
+    shift = -int(np.argmax(prof_data))+int(len(prof_data))//2
+    prof_data = np.roll(prof_data, shift)
+
+    #find std and check if profile is scattered
+    sigma, flags = sigmaClip(prof_data, tol=0.01, ntrials=100, alpha=alpha)
+    check_clip(flags)
+    bot_prof_min = (max(prof_data) - min(prof_data)) * .1 + min(prof_data)
+    scattered=False
+    if (np.nanmin(flags) > bot_prof_min) or ( not np.isnan(flags).any() ):
+        logger.warning("The profile is highly scattered. S/N estimate cannot be calculated")
+        scattered=True
+        sn = sn_e = None
+    else:
+        prof_e = 500. #this is an approximation
+        non_pulse_bins = 0
+        #work out the above parameters
+        for i, data in enumerate(prof_data):
+            if not np.isnan(flags[i]):
+                non_pulse_bins += 1
+        sigma_e = sigma / np.sqrt(2 * non_pulse_bins - 2)
+        #now calc S/N
+        sn = max(prof_data)/sigma
+        sn_e = sn * np.sqrt(prof_e/max(prof_data)**2 + (sigma_e/sigma)**2)
+
+    return [sn, sn_e, scattered]
+
+#---------------------------------------------------------------
+def integral_multi_gauss(*params):
+    y=0
+    for i in range(0, len(params), 3):
+        a = params[i]
+        b = params[i+1]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused variable 'b'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)